build/source/engine/vegNrgFlux.f90

```diff
@@ -2191,66 +2187,49 @@ subroutine aeroResist(&
   ! compute the above-canopy resistance (s m-1)
   canopyResistance = 1._dp/(sfc2AtmExchangeCoeff_canopy*windspd)
   if(canopyResistance < 0._dp)then; err=20; message=trim(message)//'canopy resistance < 0'; return; end if
-  !write(*,'(a,10(f20.10,1x))') 'in aeroResist: windspd, canairTemp, canopyExNeut, canopyStabilityCorrection, canopyResistance = ', &
-  !                                             windspd, canairTemp, canopyExNeut, canopyStabilityCorrection, canopyResistance
 
   ! compute windspeed at the top of the canopy (m s-1)
   ! NOTE: stability corrections cancel out
-  windConvFactorTop = log((heightCanopyTop - zeroPlaneDisplacement)/z0Canopy) / log((mHeight - zeroPlaneDisplacement)/z0Canopy)
+  windConvFactorTop = log((heightCanopyTopAboveSnow - zeroPlaneDisplacement)/z0Canopy) / log((mHeight - snowDepth - zeroPlaneDisplacement)/z0Canopy)
   windspdCanopyTop  = windspd*windConvFactorTop
-  !windConvFactorTop = log((heightCanopyTop - zeroPlaneDisplacement)/z0Canopy)/(sqrt(canopyStabilityCorrection)*vkc)
-  !windspdCanopyTop  = frictionVelocity*windConvFactorTop
 
   ! compute the windspeed reduction
   ! Refs: Norman et al. (Ag. Forest Met., 1995) -- citing Goudriaan (1977 manuscript "crop micrometeorology: a simulation study", Wageningen).
-  windReductionFactor = windReductionParam * exposedVAI**twoThirds * (heightCanopyTop - heightCanopyBottom)**oneThird / leafDimension**oneThird
-  !windReductionFactor = 3._dp
+  windReductionFactor = windReductionParam * exposedVAI**twoThirds * (heightCanopyTopAboveSnow - heightCanopyBottomAboveSnow)**oneThird / leafDimension**oneThird
 
   ! compute windspeed at the bottom of the canopy (m s-1)
-  !referenceHeight      = max(heightCanopyBottom, min(0.5_dp, heightCanopyTop))
-  referenceHeight      = max(heightCanopyBottom, snowDepth+z0Ground)
-  windConvFactorBottom = exp(-windReductionFactor*(1._dp - referenceHeight/heightCanopyTop))
+  referenceHeight      = max(heightCanopyBottomAboveSnow, z0Ground)
+  windConvFactorBottom = exp(-windReductionFactor*(1._dp - (referenceHeight/heightCanopyTopAboveSnow)))
```

Comment:
There calculations are only used when the snow is below the bottom of the canopy. Could you move these calculations to the `case(logBelowCanopy)` below?

Reply:
There is a call to `aStability` which uses `windspdCanopyBottom`, so we actually can't move these inside the `select` statement.